### PR TITLE
Factorized bulk operations

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Signed-off-by: Kevin Boulongne <kevin.boulongne@infomaniak.com>

Closes: #600

### CODE REVIEW

File: app/src/main/java/com/infomaniak/drive/ui/fileList/multiSelect/MultiSelectFragment.kt [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+            putExtra(USER_DRIVE_ID_TAG, AccountUtils.currentDriveId)
+            putExtra(CUSTOM_ARGS_TAG, bundleOf(BULK_OPERATION_CUSTOM_TAG to BulkOperationType.COPY))
+        }
+        selectFolderResultLauncher.launch(intent)
+    }
+
+    open fun performBulkOperation(
+        type: BulkOperationType,
+        areAllFromTheSameFolder: Boolean = true,
+        selectedFilesCount: Int? = null,
```
Comment (JorisBodin) [inline app/src/main/java/com/infomaniak/drive/ui/fileList/multiSelect/MultiSelectFragment.kt:161]: You have to explain to me what it is used for, knowing that after we call `getValidSelectedItems `
The only use I see is for the "all selected", so at that point rename it everywhere.

**Review by JorisBodin — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/drive/ui/fileList/multiSelect/MultiSelectFragment.kt [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+
+    fun onItemSelected(selectedNumber: Int? = null) {
+        val fileSelectedNumber = selectedNumber ?: multiSelectManager.getValidSelectedItems().size
+
+        when (fileSelectedNumber) {
+            0, 1 -> {
+                val isEnabled = fileSelectedNumber == 1
+                enableMultiSelectButtons(isEnabled)
+            }
+        }
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/drive/ui/fileList/multiSelect/MultiSelectFragment.kt:115]:
```suggestion
        if (fileSelectedNumber in 0..1) {
            val isEnabled = fileSelectedNumber == 1
            enableMultiSelectButtons(isEnabled)
        }
```

File: app/src/main/java/com/infomaniak/drive/ui/fileList/multiSelect/MultiSelectFragment.kt [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+        multiSelectLayout?.root?.isVisible = true
+    }
+
+    fun onItemSelected(selectedNumber: Int? = null) {
+        val fileSelectedNumber = selectedNumber ?: multiSelectManager.getValidSelectedItems().size
+
+        when (fileSelectedNumber) {
+            0, 1 -> {
+                val isEnabled = fileSelectedNumber == 1
+                enableMultiSelectButtons(isEnabled)
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/drive/ui/fileList/multiSelect/MultiSelectFragment.kt]:
```suggestion
                enableMultiSelectButtons(fileSelectedNumber == 1)
```

File: app/src/main/java/com/infomaniak/drive/ui/fileList/multiSelect/FileListMultiSelectActionsBottomSheetDialog.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+                isVisible = true
+            }
+        }
+    }
+
+    private fun computeColoredFolderAvailability(fileIds: IntArray): Boolean {
+        fileIds.forEach {
+            val file = FileController.getFileById(it)
+            if (file?.isAllowedToBeColored() == true) return true
+        }
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/drive/ui/fileList/multiSelect/FileListMultiSelectActionsBottomSheetDialog.kt:41]: You should avoid creating several instances of realm in a row like this

**Review by JorisBodin — APPROVED** [review]

**Review by sirambd — APPROVED** [review]

**Pull request merged**